x-pack/plugins/apm/public/components/alerting/ui_components/apm_rule_unified_search_bar.tsx

```diff
@@ -9,7 +9,7 @@ import { i18n } from '@kbn/i18n';
 import { Query } from '@kbn/es-query';
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import { ApmPluginStartDeps } from '../../../plugin';
-import { useApmDataView } from '../../../hooks/use_apm_data_view';
+import { useAdHocApmDataView } from '../../../hooks/use_ad_hoc_apm_data_view';
```

Comment:
Ah, so we were already using "ad hoc" data views in several places, we just didn't call it as such?

Reply:
Yes,  that makes me think though.  
Do we need to create an ad hoc data view every single time we use it or would it be better to create it once and pass it in the context?